File: backend/app/schemas/dentist.py
"""
OraVisionAI — Dentist Pydantic Schemas

Data transfer schemas for dentist professional profiles and verification document submissions.
"""

from __future__ import annotations

import datetime
import uuid
from typing import Optional

from pydantic import BaseModel, ConfigDict, Field


class DentistBase(BaseModel):
    specialization: Optional[str] = Field("General Dentistry", max_length=150)
    clinic_name: Optional[str] = Field(None, max_length=200)
    clinic_address: Optional[str] = None
    years_of_experience: Optional[int] = Field(0, ge=0)
    bio: Optional[str] = None


class DentistUpdate(BaseModel):
    specialization: Optional[str] = Field(None, max_length=150)
    clinic_name: Optional[str] = Field(None, max_length=200)
    clinic_address: Optional[str] = None
    years_of_experience: Optional[int] = Field(None, ge=0)
    bio: Optional[str] = None
    first_name: Optional[str] = Field(None, min_length=1, max_length=100)
    last_name: Optional[str] = Field(None, min_length=1, max_length=100)
    phone_number: Optional[str] = Field(None, max_length=30)
    avatar_url: Optional[str] = None

    model_config = ConfigDict(extra="forbid")


class DentistResponse(BaseModel):
    id: uuid.UUID
    user_id: uuid.UUID
    email: str
    first_name: str
    last_name: str
    phone_number: Optional[str] = None
    avatar_url: Optional[str] = None
    license_number: str
    specialization: str
    clinic_name: Optional[str] = None
    clinic_address: Optional[str] = None
    years_of_experience: int
    bio: Optional[str] = None
    verification_status: str
    verified_at: Optional[datetime.datetime] = None
    rejection_reason: Optional[str] = None
    created_at: datetime.datetime
    updated_at: datetime.datetime

    model_config = ConfigDict(from_attributes=True)


class DentistVerificationBase(BaseModel):
    document_type: str = Field(..., max_length=100)
    document_url: str = Field(...)
    file_name: str = Field(..., max_length=255)
    file_size_bytes: Optional[int] = Field(None, ge=0)


class DentistVerificationCreate(DentistVerificationBase):
    model_config = ConfigDict(extra="forbid")


class DentistVerificationResponse(DentistVerificationBase):
    id: uuid.UUID
    dentist_id: uuid.UUID
    status: str
    submitted_at: datetime.datetime
    reviewed_at: Optional[datetime.datetime] = None
    review_notes: Optional[str] = None

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/dentist.ts
/**
 * OraVisionAI — Dentist, Appointment & Assessment Domain Types
 *
 * Derived strictly from the frozen Phase 21 backend Pydantic schemas:
 * - app/schemas/dentist.py
 * - app/schemas/appointment.py
 * - app/schemas/dentist_assessment.py
 */

import { DentistVerificationStatus } from './domain';

// ============================================================================
// 1. Dentist Profile & Verification
// ============================================================================

export interface DentistProfile {
  id: string;
  user_id: string;
  email: string;
  first_name: string;
  last_name: string;
  phone_number?: string | null;
  avatar_url?: string | null;
  license_number: string;
  specialization: string;
  clinic_name?: string | null;
  clinic_address?: string | null;
  years_of_experience: number;
  bio?: string | null;
  verification_status: DentistVerificationStatus;
  verified_at?: string | null;
  rejection_reason?: string | null;
  created_at: string;
  updated_at: string;
}

export interface DentistProfileUpdate {
  specialization?: string;
  clinic_name?: string | null;
  clinic_address?: string | null;
  years_of_experience?: number;
  bio?: string | null;
  first_name?: string;
  last_name?: string;
  phone_number?: string | null;
  avatar_url?: string | null;
}

export interface DentistVerification {
  id: string;
  dentist_id: string;
  document_type: string;
  document_url: string;
  file_name: string;
  file_size_bytes?: number | null;
  status: DentistVerificationStatus;
  submitted_at: string;
  reviewed_at?: string | null;
  review_notes?: string | null;
}

export interface DentistVerificationCreate {
  document_type: string;
  document_url: string;
  file_name: string;
  file_size_bytes?: number | null;
}

// ============================================================================
// 2. Appointment Lifecycle & Scheduling
// ============================================================================

export type AppointmentStatus =
  | 'requested'
  | 'confirmed'
  | 'in_progress'
  | 'completed'
  | 'cancelled'
  | 'rescheduled'
  | 'no_show';

export type AppointmentType =
  | 'video_teleconsultation'
  | 'audio_teleconsultation'
  | 'in_person_consultation'
  | 'follow_up';

export interface Appointment {
  id: string;
  patient_id: string;
  dentist_id: string;
  screening_id?: string | null;
  scheduled_start: string;
  scheduled_end: string;
  appointment_type: AppointmentType | string;
  status: AppointmentStatus | string;
  cancellation_reason?: string | null;
  cancelled_by_id?: string | null;
  patient_notes?: string | null;
  dentist_notes?: string | null;
  created_at: string;
  updated_at: string;
  patient_name?: string | null;
  dentist_name?: string | null;
  clinic_name?: string | null;
}

export interface AppointmentListResponse {
  items: Appointment[];
  total: number;
}

export interface AppointmentStatusUpdate {
  status: AppointmentStatus;
  dentist_notes?: string | null;
}

export interface AppointmentCancel {
  cancellation_reason: string;
}

// ============================================================================
// 3. Clinical Assessment & Screening Evaluation
// ============================================================================

export interface DentistAssessment {
  id: string;
  screening_id: string;
  dentist_id: string;
  clinical_observations: string;
  diagnosis_notes: string;
  treatment_recommendation: string;
  referral_needed: boolean;
  referral_specialty?: string | null;
  is_finalized: boolean;
  finalized_at?: string | null;
  created_at: string;
  updated_at: string;
  dentist_name?: string | null;
  dentist_clinic?: string | null;
  dentist_license?: string | null;
}

export interface DentistAssessmentCreate {
  clinical_observations: string;
  diagnosis_notes: string;
  treatment_recommendation: string;
  referral_needed?: boolean;
  referral_specialty?: string | null;
  is_finalized?: boolean;
}

export interface DentistAssessmentUpdate {
  clinical_observations?: string;
  diagnosis_notes?: string;
  treatment_recommendation?: string;
  referral_needed?: boolean;
  referral_specialty?: string | null;
  is_finalized?: boolean;
}
